docs/firefighting/index.md

```diff
@@ -0,0 +1,788 @@
+
+If disk usage on the proxy ever hits 100%, many basic functions on the machine will stop working.  Here are some ways to buy some time.
+
+## Clean Releases
+Each release / copy of our commcare-hq git repo can be 500M - 2.7G (variation can depend on how efficiently git is storing the history, etc.). It's always safe to run
+
+`$ cchq <env> fab clean_releases`
+and sometimes that alone can clean up space. This is run on every deploy, so if you just deployed successfully, don't bother.
+
+## Move logs to another drive
+Check the size of the log files stored at /home/cchq/www/<environment>/log, these can get out of hand.  Last time this ocurred, we moved these into the shared drive, which had plenty of available disk space (but check first!)
+
+`$ mv -v pattern-matching-old-logs.*.gz /mnt/shared/temp-log-storage-main-hd-full/`
+
+## Clear the apt cache
+`$ apt-get autoremove`
+On a recent occasion, this freed up about 20% of the disk
+
+## Manually rotate syslog
+If for some reason syslog is either not rotating logs or the latest log has grown more than expected you can run
+
+```
+mv syslog other.syslog
+kill -HUP <pid of syslog>
+gzip other.syslog
+```
+
+
+# Network Connection Issues (please help expand)
+
+
+If you suspect some sort of network issue is preventing two servers from talking to each other, the first thing you should verify is that the processes you expect to talk to each other are actually running.  After that, here are some things to try:
+
+## Ping
+Try pinging the machine from the computer you'd expect to initiate the connection.  If that doesn't work, try pinging from your local machine while on the VPN.
+
+```
+esoergel@hqproxy0:~$ ping hqdb0.internal-va.commcarehq.org
+PING hqdb0.internal-va.commcarehq.org (172.24.32.11) 56(84) bytes of data.
+64 bytes from 172.24.32.11 (172.24.32.11): icmp_seq=1 ttl=64 time=42.6 ms
+64 bytes from 172.24.32.11 (172.24.32.11): icmp_seq=2 ttl=64 time=41.3 ms
+```
+
+## netcat
+Netcat is a mini server.  You can set it up to listen on any port and respond to requests.  Run something like this to listen on port 1234 and wait for a request:
+
+`esoergel@hqdb0:~$ printf "Can you see this?" | netcat -l 1234`
+
+Then go over to the other machine and try to hit that server:
+
+```
+$ curl hqdb0.internal-va.commcarehq.org:1234
+Can you see this?$
+```
+
+Looks like the request went through!  If you go back and check on the netcat process, you should see the request:
+
+```
+esoergel@hqdb0:~$ printf "Can you see this?" | netcat -l 1234
+HEAD / HTTP/1.1
+Host: hqdb0.internal-va.commcarehq.org:1234
+User-Agent: curl/7.50.1
+Accept: */*
+
+esoergel@hqdb0:~$
+```
+
+# Tips and Tricks
+Never run that painful sequence of sudo -u cchq bash, entering the venv, cd'ing to the directory, etc., again just to run a management command. Instead, just run e.g.:
+
+`sudo -u cchq bash -c 'cd /home/cchq/www/production/current && /home/cchq/www/production/current/python_env/bin/python manage.py shell'i`
```

Comment:
nit: trailing `i` looks like a mistake.

Wonder if it would be useful to add `cchq <env> django-manage shell` in here since that's essentially equivalent (except it's always executed on the `django_manage` machine)?

Reply:
I'd be in favor of adding that (though that sounds like a separate change, out of scope of this transfer from confluence).  I do find this snippet very useful though, such as when I'm already working on a specific machine, so I'd want that as an addition, not a replacement.

Reply:
There were a lot of these sorts of changes I noticed, but wanted to keep it small as possible. I only changed to  the cchq commands for the big services table because that was easier than transferring that table to markdown.